### PR TITLE
Add msmtp 1.8.7 port.

### PR DESCRIPTION
**Pull request by haydenroche5**

### CODE REVIEW

**Comment by haydenroche5** [comment]
There are no unit tests for msmtp, at least that I can find. I set up a dummy email account and was able to send mail using it, though. The instructions for doing that are a bit too involved to be included in the README, though, in my opinion.

File: msmtp/1.8.7/wolfssl-msmtp-1.8.7.patch [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
++	AC_MSG_ERROR([Use --with-tls=gnutls or --with-tls=openssl or --with-tls=wolfssl or --with-tls=no])
+     fi)
+ if test "$want_gnutls" = "yes"; then
+     PKG_CHECK_MODULES([libgnutls], [gnutls >= 0.0], [HAVE_LIBGNUTLS=1], [HAVE_LIBGNUTLS=0])
+@@ -147,6 +156,21 @@ if test "$want_openssl" = "yes" -a "$have_tls" = "no"; then
+ 	AC_DEFINE([HAVE_LIBSSL], [1], [Define to 1 if libssl is available])
+     fi
+ fi
++if test "$want_wolfssl" = "yes" -a "$have_tls" = "no"; then
++    AC_MSG_WARN([Using OpenSSL is discouraged; consider using GnuTLS instead])
```
Comment (cconlon) [inline msmtp/1.8.7/wolfssl-msmtp-1.8.7.patch:48]: Is this AC_MSG_WARN accurate?  Looks like this is in the wolfSSL section, but mentions OpenSSL and GnuTLS.

**Review by cconlon — CHANGES_REQUESTED** [review]
One minor comment/question.

File: msmtp/1.8.7/wolfssl-msmtp-1.8.7.patch (hunk 1 above)
Comment (haydenroche5) [reply msmtp/1.8.7/wolfssl-msmtp-1.8.7.patch:48]: Nice catch. I appear to have left this in by accident after copy-pasting from the OpenSSL stanza.

**Review by cconlon — APPROVED** [review]

**Pull request merged**